modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/DateTimeUtil.java

```diff
@@ -0,0 +1,217 @@
+ */
+public class DateTimeUtil {
+
+	public static Date toDate(String value) {
+		return toDate(toZonedDateTime(value));
+	}
+
+	public static Date toDate(ZonedDateTime zonedDateTime) {
+		if (zonedDateTime == null) {
+			return null;
+		}
+
+		return Date.from(zonedDateTime.toInstant());
+	}
+
+	public static LocalDateTime toLocalDateTime(Date date) {
+		return toLocalDateTime(date, null);
+	}
+
+	public static LocalDateTime toLocalDateTime(Date date, Date defaultDate) {
+		return toLocalDateTime(date, defaultDate, ZoneId.systemDefault());
+	}
+
+	public static LocalDateTime toLocalDateTime(
+		Date date, Date defaultDate, ZoneId zoneId) {
+
+		Instant instant = null;
+
+		if (date == null) {
+			if (defaultDate == null) {
+				defaultDate = new Date();
+			}
+
+			instant = defaultDate.toInstant();
+		}
+		else {
+			instant = date.toInstant();
+		}
+
+		ZonedDateTime zonedDateTime = instant.atZone(zoneId);
+
+		return zonedDateTime.toLocalDateTime();
+	}
+
+	public static LocalDateTime toLocalDateTime(String value) {
+		return toZonedDateTime(
+			null, value
+		).toLocalDateTime();
+	}
+
+	public static LocalDateTime toLocalDateTime(
+		String timeZoneId, String value) {
+
+		return toZonedDateTime(
+			timeZoneId, value
+		).toLocalDateTime();
+	}
+
+	public static Timestamp toTimestamp(String timeZoneId, String value) {
+		return toTimestamp(toZonedDateTime(timeZoneId, value));
+	}
+
+	public static Timestamp toTimestamp(ZonedDateTime zonedDateTime) {
+		return Timestamp.from(zonedDateTime.toInstant());
+	}
+
+	public static ZonedDateTime toZonedDateTime(String value) {
+		return toZonedDateTime(null, value);
+	}
+
+	public static ZonedDateTime toZonedDateTime(
+		String timeZoneId, String value) {
+
+		String dateTimePattern = _getDateTimePattern(value);
+
+		if (dateTimePattern == null) {
+			throw new IllegalArgumentException(
+				"Unable to parse date-time from " + value);
+		}
+
+		DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(
+			dateTimePattern);
+
+		if (dateTimePattern.length() == 10) {
+			LocalDateTime localDateTime = LocalDate.parse(
+				value, dateTimeFormatter
+			).atStartOfDay();
+
+			return localDateTime.atZone(ZoneId.of(StringPool.UTC));
+		}
+
+		if (dateTimePattern.contains("XXX") ||
+			dateTimePattern.contains("zzz")) {
+
+			try {
+				return ZonedDateTime.parse(value, dateTimeFormatter);
+			}
+			catch (DateTimeParseException dateTimeParseException) {
+				throw new RuntimeException(dateTimeParseException);
+			}
+		}
+
+		try {
+			LocalDateTime localDateTime = LocalDateTime.parse(
+				value, dateTimeFormatter);
+
+			if (timeZoneId == null) {
+				return localDateTime.atZone(ZoneId.of(StringPool.UTC));
+			}
+
+			return localDateTime.atZone(ZoneId.of(timeZoneId));
+		}
+		catch (DateTimeParseException dateTimeParseException) {
+			throw new RuntimeException(dateTimeParseException);
+		}
+	}
+
+	private static String _getDateTimePattern(String value) {
+		if (Validator.isNull(value)) {
```

Comment:
I think this could be simplified a bit. Testing this and no need to check for length > 10 later:
```
		if (Validator.isNull(value) || value.length() < 10 ) {
			return null;
		}
```